### ISSUE TITLE
Upgrade ASM from 1.4 to 1.6+

### ISSUE THREAD

**Issue by jlewi** [kubeflow/gcp-blueprints#144]
We are currently using ASM 1.4. Docs currently recommend 1.6.
https://cloud.google.com/service-mesh/docs/gke-install-asm

I don't think 1.4 is still supported.

kubeflow/kubeflow#5313 is tracking ISTIO 1.6 in Kubeflow

**Comment by Bobgy** [comment]
Agree, added to https://github.com/kubeflow/gcp-blueprints/issues/139.
We need to confirm the CRDs we are relying on, like Istio RBAC are still supported on 1.6.

**Comment by Bobgy** [comment]
I think I didn't realize how important this is, the fact that ASM 1.4 is no longer supported make this a high priority for us to migrate.

**Comment by Bobgy** [comment]
Documentation to refer:
https://cloud.google.com/service-mesh/docs/scripted-install/asm-onboarding
https://cloud.google.com/service-mesh/docs/enable-optional-features

**Comment by Bobgy** [comment]
Official doc to refer: https://istio.io/latest/docs/concepts/what-is-istio/

**Comment by zijianjoy** [comment]
Latest istioctl-asm download page: http://cloud/service-mesh/docs/downloading-istioctl#mac-os

**Comment by zijianjoy** [comment]
Attempted to run `make apply` directly after upgrading istioctl-asm. See the following error for command:

command:
`istioctl manifest generate -f $(MANIFESTS_DIR)/gcp/v2/asm/istio-operator.yaml -o $(BUILD_DIR)/istio`

error:

```
#************************************************************************************
# hydrate asm
istioctl manifest generate -f ./upstream/manifests/gcp/v2/asm/istio-operator.yaml -o .build/istio
Error: validation errors (use --force to override):
unknown field "env" in v1alpha1.ProxyConfig
make: *** [hydrate-asm] Error 1
```

**Comment by zijianjoy** [comment]
Exploring applying https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/master/asm/istio/istio-operator.yaml

**Comment by zijianjoy** [comment]
Making changes in `Makefile` to simulate manual steps in https://cloud.google.com/service-mesh/docs/scripted-install/gke-install.

Currently facing issues:

*  `sni_hosts` are not recognized because validation has been enforced on this field. Changing them to `sniHosts` in `kf-istio-resources.html` fixes the issue.

*  There is no match for `ClusterRbacConfig` in "rbac.istio.io/v1alpha1", because it has been deprecated. Reading https://istio.io/latest/blog/2019/v1beta1-authorization-policy/ to learn more.

**Comment by zijianjoy** [comment]
Need to follow up with csm lifecycle team about how to maintain `ASM revision` namespace label because its version can get deprecated easily.

**Comment by zijianjoy** [comment]
We also need to document the steps to upgrade ASM if user wants to.

**Comment by rky0930** [comment]
@zijianjoy If Kubeflow 1.3 uses ASM1.9, it would be better if 1.9.3-asm.2 or higher is used by default. This is because a security issue, [ISTIO-SECURITY-2021-003](https://istio.io/latest/news/security/istio-security-2021-003/), was recently resolved in version 1.9.3-asm.2. Please refer to the [Release Note](https://cloud.google.com/service-mesh/docs/release-notes) here.

**Comment by zijianjoy** [comment]
> @zijianjoy If Kubeflow 1.3 uses ASM1.9, it would be better if 1.9.3-asm.2 or higher is used by default. This is because a security issue, [ISTIO-SECURITY-2021-003](https://istio.io/latest/news/security/istio-security-2021-003/), was recently resolved in version 1.9.3-asm.2. Please refer to the [Release Note](https://cloud.google.com/service-mesh/docs/release-notes) here.

Thank you @rky0930 ! I am also working on the documentation to allow users to upgrade ASM to any version by themselves. Once I send out the PR for documentation, would you like to give it a try to see if your upgrade works? Because I expect there will be future ASM/istio releases, so I think it will benefit in a long term if the cluster maintainer knows how to upgrade ASM, instead of relying on Kubeflow release schedule which is always behind the latest.

**Comment by rky0930** [comment]
@zijianjoy Okay, when the document PR is created, I'll try it. Thanks !

**Comment by zijianjoy** [comment]
Thank you so much @rky0930!

**Comment by zijianjoy** [comment]
@rky0930 Hello Gyoung-yoon, we have released Kubeflow 1.3 on Google Cloud and you can refer to https://www.kubeflow.org/docs/distributions/gke/deploy/overview/ for the installation.

As part of this release, I have upgraded ASM to the latest available version v1.9.3-2. So the security issue mentioned by you should be covered. For future ASM release, you can refer to https://www.kubeflow.org/docs/distributions/gke/deploy/upgrade/#upgrade-asm-anthos-service-mesh to upgrade ASM in your Kubeflow cluster. Feedback on the deployment and upgrade processes are welcomed!
